src/main/java/org/jgroups/protocols/shm/SHM.java

```diff
@@ -145,10 +157,14 @@ public void sendUnicast(PhysicalAddress dest, byte[] data, int offset, int lengt
     }
 
     @Override
-    public void accept(ByteBuffer bb, Integer length) {
+    public void accept(ByteBuffer bb) {
         try {
-            DataInput in=new ByteBufferInputStream(bb);
-            receive(null, in);
+            LeakyByteBufferInputStream receiveStream = this.cachedReceiveStream;
+            if (receiveStream == null || receiveStream.buf() != bb) {
+                receiveStream = new LeakyByteBufferInputStream(bb);
+                this.cachedReceiveStream = null;
+            }
+            receive(null, receiveStream);
```

Comment:
Important: `receive` cannot close it or we need to modify `LeakyByteBufferInputStream` to open it again while reusing it

Reply:
@franz1981 receive() does never close the input stream; this is JGroups code